[udev] Added udev rule for GC adapter

aserna3 opened the pull request:
This fixes a blocker: https://bugs.dolphin-emu.org/issues/8664

The previous PR gave generous permissions to the udev rule which many didn't feel was ideal. This uses a built in udev function many modern distros have, which should work fine.

Note: I do not have a GC adapter and cannot test this. Somebody test this please. :3

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/dolphin-emu/dolphin/3670)
<!-- Reviewable:end -->

Antidote commented on Installer/51-gcadapter.rules:
```diff
@@ -0,0 +1 @@
++SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ATTRS{idVendor}=="057e", ATTRS{idProduct}=="0337", TAG+="uaccess"
```
Superfluous '+' breaks the rule, otherwise it works

delroth commented:
1. I still don't believe we should auto-install udev rules, let it to the distros to do that (and provide them with documentation).
2. Current install doesn't respect the install prefix.
3. Putting stuff like that in the "Installer/" directory is just weird.

aserna3 commented:
I agree with the ideal option not being automatically installing the file. I've removed it from the cmake script. I've also moved the file from Installer/ to Data/ and updated README.md with details on the new file. I don't know how best to put that new documentation so feel free to pick at my wording / placement with something better.

rukai commented:
Reviewed 3 of 4 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3670)*
<!-- Sent from Reviewable.io -->

rukai commented:
I have tested the 51-gcadapter.rules file and it works fine.

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3670#-:-KC8MxXUx_5Dlgqb8kj0:1266960891)*
<!-- Sent from Reviewable.io -->

BhaaLseN commented on Readme.md:
```diff
@@ -155,6 +155,12 @@ These folders are installed read-only and should not be changed:
 * `Wii`: default Wii NAND contents
 
+## Packaging and udev
+In the Data folder there is a udev rule file for official GameCube controller
```
Maybe start with "The Data folder contains a udev rule file..." instead? For a non-native speaker plagued by german colleagues literally translating things, this sounds too germanglished to me...

aserna3 replied:
Done

delroth commented on Data/51-gcadapter.rules:
```diff
@@ -0,0 +1 @@
+SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ATTRS{idVendor}=="057e", ATTRS{idProduct}=="0337", TAG+="uaccess"
```
Missing a final \n

aserna3 replied:
Done

degasus commented:
Shall we also include the dolphin bar? We still lack linux support, but in my opinion, it's good to already have the permission set for it.

aserna3 commented:
@degasus what is the USB id for the Dolphinbar? I don't see why not.

aserna3 commented:
If somebody could give me the USB ID for DolphinBar I'll add it in and then this PR will be ready to merge

aserna3 commented:
Done.
